glTFLoader/Interface.cs

```diff
@@ -71,55 +79,160 @@ private static string ParseBinary(Stream stream)
+                }
+
+                uint chunkFormat = binaryReader.ReadUInt32();
+
+                var data = binaryReader.ReadBytes((int)chunkLength);
+
+                if (chunkFormat == format) return data;                
+            }            
+        }        
+
+        public static Byte[] LoadBinaryBuffer(string filePath)
+        {
+            using (Stream stream = File.OpenRead(filePath))
+            {
+                return LoadBinaryBuffer(stream);
+            }
+        }
+
+        public static Byte[] LoadBinaryBuffer(Stream stream)
+        {
+            using (BinaryReader binaryReader = new BinaryReader(stream))
+            {
+                ReadBinaryHeader(binaryReader);
 
-                return Encoding.UTF8.GetString(binaryReader.ReadBytes((int)chunkLength));
+                return ReadBinaryChunk(binaryReader, BIN);
             }
         }
 
+        private static void ReadBinaryHeader(BinaryReader binaryReader)
+        {
+            uint magic = binaryReader.ReadUInt32();
+            if (magic != GLTF)
+            {
+                throw new InvalidDataException($"Unexpected magic number: {magic}");
+            }
+
+            uint version = binaryReader.ReadUInt32();
+            if (version != VERSION2)
+            {
+                throw new InvalidDataException($"Unknown version number: {version}");
+            }
+
+            uint length = binaryReader.ReadUInt32();
+            long fileLength = binaryReader.BaseStream.Length;
+            if (length != fileLength)
+            {
+                throw new InvalidDataException($"The specified length of the file ({length}) is not equal to the actual length of the file ({fileLength}).");
+            }
+        }
+
+        private static Func<string,Byte[]> GetExternalFileSolver(string gltfFilePath)
+        {
+            return asset =>
+            {
+                if (asset == null) return LoadBinaryBuffer(gltfFilePath);
+                var bufferFilePath = Path.Combine(Path.GetDirectoryName(gltfFilePath), asset);
+                return File.ReadAllBytes(bufferFilePath);
+            };
+        }
+
+        public static Byte[] LoadBinaryBuffer(this Gltf model, string gltfFilePath, int bufferIndex)
+        {
+            return LoadBinaryBuffer(model, GetExternalFileSolver(gltfFilePath), bufferIndex);            
+        }
+
+        public static Stream OpenImageFile(this Gltf model, string gltfFilePath, int imageIndex)
+        {
+            return OpenImageFile(model, GetExternalFileSolver(gltfFilePath), imageIndex);
+        }
+
+        public static Byte[] LoadBinaryBuffer(this Gltf model, Func<string,Byte[]> externalReferenceSolver, int bufferIndex)
+        {
+            var buffer = model.Buffers[bufferIndex];
+
+            if (buffer.Uri == null) return externalReferenceSolver(null);
+
+            if (buffer.Uri.StartsWith(EMBEDDEDOCTETSTREAM))

```

Comment:
While spec says that binary buffers use `application/octet-stream`, some tools may start using `application/gltf-buffer` after IANA approval. See https://github.com/KhronosGroup/glTF/issues/944.

Reply:
We can fix this once it's updated.

Reply:
Yeah, it's something easy to add